README.md

```diff
@@ -201,6 +201,219 @@ hints and first-parties can use `Feature-Policy` to decide which third-parties t
 privilege with detailed user agent information. In any event, moving to this opt-in model means
 that the extent of a site's usage can be monitored and evaluated.
 
+# Use-cases
+
+This section attempts to document the current uses for the `User-Agent` string,
+and how similar functionality could be enabled using UA-CH.
+
+## Differential serving based on browser features
+This use-case enables services like [polyfill.io](https://polyfill.io) to serve
+custom-tailored polyfills to their users, without bloating up the experience of
+modern browser users.
+Similarly, when serving Javascript to users, one can avoid
+transpilation (which can result in bloat and inefficient code) for browsers
+that support the latest ES features that were used.
+Finally, when serving images, some browsers don't update their `Accept` request
+headers, while in other cases (*cough* WebP *cough*) the MIME type is not
+descriptive enough to distinguish between different variants of the same
+format. In those cases, knowing the browser and its version can be critical to
+serving the right image variant. 
+
+For that use case to work, the server needs to be aware of the browser and its
+meaningful version, and map that to a list of available features. That enables
+it to know which polyfill or code variant to serve.
+
+Services that wish to do that using UA-CH will need to inspect the `Sec-CH-UA`
+header, that is sent by default on every request, and modify their reponse
+based on that.
+
+## Marketshare Analytics
+A browser's market share can be extremely important. Having visibility into a
+browser's usage can encourage developers to test in that particular browser,
+ensuring fewer compatibility issues for its users. On top of that, a browser's
+market share can have a direct impact on the browser vendors' business goals,
+ensuring future development of the browser.
+
+For marketshare analytics to work, the server needs to be aware of the server
+and its meaningful version, in order to be able to register them and find their
+relative market shares.
+
+Sites that wish to provide market share analytics using UA-CH will need to
+inspect the `Sec-CH-UA` header, that is sent by default on every request, and
```

Comment:
Again, we're not certain the accurate browser should be sent by default on every request. Here one of the options is to send the accurate browser on a fixed fraction of requests, and have the logs analysis divide by that fraction.